BenchPress/Generators/AzureDeploymentImporter.cs

```diff
@@ -229,4 +205,122 @@ string outputFolderPath
 
         return extraProperties;
     }
+
+    /// <summary>
+    /// Takes a string from an ARM template containing instances of <c>parameters('...')</c> and
+    /// <c>variables('...')</c> and resolves those instances to the correct values, if possible.
+    /// <example>
+    /// For example, if the following ARM template (<paramref name="armTemplateObject"/>) with the parameter block
+    /// below is passed in:
+    /// <code>
+    /// {...
+    ///   "parameters": {
+    ///     "demoParam": {
+    ///       "type": "string",
+    ///       "defaultValue": "Contoso"
+    ///     }
+    ///   }
+    /// ...}
+    /// </code>
+    /// and the following string (<paramref name ="stringToResolve"/>) is passed in:
+    /// <code>
+    /// "[parameters('demoParam')]"
+    /// </code>
+    /// The parameter will be resolved to the default value and result in the following return value:
+    /// <code>
+    /// "Contoso"
+    /// </code>
+    /// This method can also handle resolving ARM template variables. For example if the following ARM template
+    /// (<paramref name="armTemplateObject"/>) with the variable block below is passed in:
+    /// <code>
+    /// {...
+    ///   "variables": {
+    ///     "demoVar": "Contoso"
+    ///   }
+    /// ...}
+    /// </code>
+    /// and the following string (<paramref name ="stringToResolve"/>) is passed in:
+    /// <code>
+    /// "[variables('demoVar')]"
+    /// </code>
+    /// The variable will be resolved to the correct value and result in the following return value:
+    /// <code>
+    /// "Contoso"
+    /// </code>
+    /// Finally, this method can also handle resolving a mixture of ARM template parameters and variables. For example
+    /// if the following ARM template (<paramref name="armTemplateObject"/>) with the parameter and variable block
+    /// below is passed in:
+    /// <code>
+    /// {...
+    ///   "parameters": {
+    ///     "demoParam": {
+    ///       "type": "string",
+    ///       "defaultValue": "ContosoParam"
+    ///     }
+    ///   }
+    ///   "variables": {
+    ///     "demoVar": "ContosoVar"
+    ///   }
+    /// ...}
+    /// </code>
+    /// and the following string (<paramref name ="stringToResolve"/>) is passed in:
+    /// <code>
+    /// "[format('{0}{1}', variables('demoVar'), parameters('demoParam')]"
+    /// </code>
+    /// The variable and parameter will be resolved to the correct values and result in the following return value:
+    /// <code>
+    /// "format('{0}{1}', "ContosoVar" , "ContosoParam")"
+    /// </code>
+    /// </summary>
+    private static string ResolveParamsAndVariables(string stringToResolve, JsonObject armTemplateObject)

```

Comment:
thoughts on moving this out to a helper class? Thinking it would make the code more testable

Reply:
I'd prefer to move it out to a helper/util class if the method is needed by other classes. For now, I'm not sure it will be. Maybe we could keep it where it is for now, and I will create a follow up issue to write unit tests for this. If it makes sense to move it out to a separate class while I write the unit tests, then I can do it then. Would that work?

Reply:
Yeah that makes sense to me!